### ISSUE TITLE
Cannot open game pass game, throws UnauthorizedAccessException

### ISSUE THREAD

**Issue by jasonmp85** [Perfare/UnityStudio#948]
I'm fairly new to all of this, but I do program for a living. I'm playing through a, uh, _foxy_ new game released this week and wished to export the manual pages from within the game to give myself a nice version of them for translation efforts, printing, etc.

I've installed it via Game Pass/the Xbox app, which means it's thrown in with all the rest of the app store binaries under `C:\Program Files\WindowsApps\`. This directory is typically unreadable by most users, so after working up the guts from reading enough tutorials to make me think it safe, I changed that entire directory's owner to the _Administrators_ group. Since I'm an admin, I can now browse it (in e.g. Windows Explorer) fine, even without _Run as Administrator_.

However, attempting to _Load Folder_ from Asset Studio (I downloaded the `net5` variant) results in the following error:

```
************** Exception Text **************
System.UnauthorizedAccessException: Access to the path 'C:\Program Files\WindowsApps\Finji.TUNIC_1.0.8.0_x64__tys0ffscxatjj\Secret Legend.exe' is denied.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at AssetStudio.AssetsManager.Load(String[] files) in C:\projects\assetstudio\AssetStudio\AssetsManager.cs:line 51
   at AssetStudio.AssetsManager.LoadFolder(String path) in C:\projects\assetstudio\AssetStudio\AssetsManager.cs:line 36
   at AssetStudioGUI.AssetStudioGUIForm.<>c__DisplayClass52_0.<loadFolder_Click>b__0() in C:\projects\assetstudio\AssetStudioGUI\AssetStudioGUIForm.cs:line 163
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at AssetStudioGUI.AssetStudioGUIForm.loadFolder_Click(Object sender, EventArgs e) in C:\projects\assetstudio\AssetStudioGUI\AssetStudioGUIForm.cs:line 163
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_0(Object state)
```

The program does this whether or not I launch it with _Run As Administrator_. The program seems to be able to open other files in that same directory tree just fine, but the `exe` itself is a problem. Its owner and group don't seem problematic, and I can launch and play the game myself just fine.

Is this expected, or am I doing something wrong?

**Comment by Perfare** [comment]
Although you have administrator rights, the exe file may be read-only, and you need to modify the permissions of this file separately.
But I think you can consider copying all files other than exe to other folders, and then operate.

**Comment by jasonmp85** [comment]
Hm, yeah I don't know these tools that well (I'm a Mac/Linux guy) and it looks like maybe the owners and permissions weren't recursing down like I expected. That all aside, I moved the `exe` files elsewhere for a bit and everything worked! Because it tried to open them, I assumed it needed them for some of its functionality, but the results seemed pretty complete once I moved the executables. Thanks!
